File: src/main/java/xyz/devcmb/treeTumblers/listeners/ListenerManager.java
package xyz.devcmb.treeTumblers.listeners;

import org.bukkit.Bukkit;
import org.bukkit.event.Listener;
import org.bukkit.plugin.PluginManager;
import xyz.devcmb.treeTumblers.TreeTumblers;

public class ListenerManager {
    public static void registerAllListeners(){
        registerListener(new PlayerAdded());
    }

    public static void registerListener(Listener listener){
        TreeTumblers plugin = TreeTumblers.getPlugin();
        PluginManager manager = Bukkit.getPluginManager();

        manager.registerEvents(listener, plugin);
    }
}

File: src/main/kotlin/xyz/devcmb/treeTumblers/listeners/ListenerManager.kt
package xyz.devcmb.treeTumblers.listeners

import org.bukkit.Bukkit
import org.bukkit.event.Listener
import org.bukkit.plugin.PluginManager
import xyz.devcmb.treeTumblers.TreeTumblers

object ListenerManager {
    fun RegisterAllListeners() {
        registerListener(PlayerConnections())
    }

    private fun registerListener(listener: Listener) {
        val pluginManager: PluginManager = Bukkit.getPluginManager()
        pluginManager.registerEvents(listener, TreeTumblers.plugin)
    }
}